### PR TITLE
Add check if user exists by name

### PR DESCRIPTION
**Pull request by MarkSalabutin**
Adds `checkIfUserExists` method to the userStorage

### CODE REVIEW

File: boltdb/user_storage.go [hunk 1]
```diff
@@ -96,6 +96,31 @@ func (us *UserStorage) UserBySocialID(id string) (model.User, error) {
...
+		}
+
+		if u := ub.Get([]byte(userID)); u == nil {
+			return ErrorNotFound
+		}
+
+		return nil
+	})
+
+	if err != nil {
```
Comment (esimonov) [inline boltdb/user_storage.go]:
```suggestion
	return err == nil
```

File: mongo/user_storage.go [hunk 2]
```diff
@@ -56,6 +63,20 @@ func (us *UserStorage) UserByFederatedID(provider model.FederatedIdentityProvide
 	return &User{userData: u}, nil
 }
 
+//CheckIfUserExistByName checks does user exist with presented name
```
[the remark this answers is not in the document]
Comment (esimonov) [reply mongo/user_storage.go]: Also, sentences usually end with period. Do not omit it please.

File: boltdb/user_storage.go [hunk 3]
```diff
@@ -92,6 +92,27 @@ func (us *UserStorage) UserByFederatedID(provider model.FederatedIdentityProvide
 	return res, nil
 }
 
+//CheckIfUserExistByName checks does user exist with presented name
+func (us *UserStorage) CheckIfUserExistByName(name string) bool {
```
Comment (erudenko) [inline boltdb/user_storage.go]: `CheckIf` is not perfect naming

File: mongo/user_storage.go [hunk 4]
```diff
@@ -183,3 +202,7 @@ func PasswordHash(pwd string) string {
 	hash, _ := bcrypt.GenerateFromPassword([]byte(pwd), bcrypt.DefaultCost)
 	return string(hash)
 }
+
+func createStrictRegex(str string) string {
```
Comment (erudenko) [inline mongo/user_storage.go]: I'm not sure it requires the dedicated function.

File: boltdb/user_storage.go [hunk 5]
```diff
@@ -92,6 +92,27 @@ func (us *UserStorage) UserByFederatedID(provider model.FederatedIdentityProvide
 	return res, nil
 }
 
+//IsNameInUse checks does user exist with presented name
+func (us *UserStorage) IsNameInUse(name string) bool {
```
Comment (erudenko) [inline boltdb/user_storage.go]: I'm not sure it's the best naming, I think, `UserExists` is the better naming

**Pull request merged**